core/primitives-core/src/parameter.rs

```diff
@@ -0,0 +1,202 @@
+    ActionDeleteKeyExecution,
+
+    // Smart contract dynamic gas costs
+    WasmRegularOpCost,
+    WasmGrowMemCost,
+    WasmHostFunctionBase,
+    WasmContractLoadingBase,
+    WasmContractLoadingBytes,
+    WasmReadMemoryBase,
+    WasmReadMemoryByte,
+    WasmWriteMemoryBase,
+    WasmWriteMemoryByte,
+    WasmReadRegisterBase,
+    WasmReadRegisterByte,
+    WasmWriteRegisterBase,
+    WasmWriteRegisterByte,
+    WasmUtf8DecodingBase,
+    WasmUtf8DecodingByte,
+    WasmUtf16DecodingBase,
+    WasmUtf16DecodingByte,
+    WasmSha256Base,
+    WasmSha256Byte,
+    WasmKeccak256Base,
+    WasmKeccak256Byte,
+    WasmKeccak512Base,
+    WasmKeccak512Byte,
+    WasmRipemd160Base,
+    WasmRipemd160Block,
+    WasmEcrecoverBase,
+    WasmLogBase,
+    WasmLogByte,
+    WasmStorageWriteBase,
+    WasmStorageWriteKeyByte,
+    WasmStorageWriteValueByte,
+    WasmStorageWriteEvictedByte,
+    WasmStorageReadBase,
+    WasmStorageReadKeyByte,
+    WasmStorageReadValueByte,
+    WasmStorageRemoveBase,
+    WasmStorageRemoveKeyByte,
+    WasmStorageRemoveRetValueByte,
+    WasmStorageHasKeyBase,
+    WasmStorageHasKeyByte,
+    WasmStorageIterCreatePrefixBase,
+    WasmStorageIterCreatePrefixByte,
+    WasmStorageIterCreateRangeBase,
+    WasmStorageIterCreateFromByte,
+    WasmStorageIterCreateToByte,
+    WasmStorageIterNextBase,
+    WasmStorageIterNextKeyByte,
+    WasmStorageIterNextValueByte,
+    WasmTouchingTrieNode,
+    WasmReadCachedTrieNode,
+    WasmPromiseAndBase,
+    WasmPromiseAndPerPromise,
+    WasmPromiseReturn,
+    WasmValidatorStakeBase,
+    WasmValidatorTotalStakeBase,
+    WasmAltBn128G1MultiexpBase,
+    WasmAltBn128G1MultiexpElement,
+    WasmAltBn128PairingCheckBase,
+    WasmAltBn128PairingCheckElement,
+    WasmAltBn128G1SumBase,
+    WasmAltBn128G1SumElement,
+
+    // Smart contract limits
+    MaxGasBurnt,
+    MaxGasBurntView,
+    MaxStackHeight,
+    StackLimiterVersion,
+    InitialMemoryPages,
+    MaxMemoryPages,
+    RegistersMemoryLimit,
+    MaxRegisterSize,
+    MaxNumberRegisters,
+    MaxNumberLogs,
+    MaxTotalLogLength,
+    MaxTotalPrepaidGas,
+    MaxActionsPerReceipt,
+    MaxNumberBytesMethodNames,
+    MaxLengthMethodName,
+    MaxArgumentsLength,
+    MaxLengthReturnedData,
+    MaxContractSize,
+    MaxTransactionSize,
+    MaxLengthStorageKey,
+    MaxLengthStorageValue,
+    MaxPromisesPerFunctionCallAction,
+    MaxNumberInputDataDependencies,
+    MaxFunctionsNumberPerContract,
+    Wasmer2StackLimit,
+    MaxLocalsPerContract,
+}
+
+pub fn load_parameters_from_txt(arg: &[u8]) -> BTreeMap<Parameter, String> {
+    let parameters = BufReader::new(arg).lines().filter_map(|line| {
+        let trimmed = line.unwrap().trim().to_owned();
+        if trimmed.starts_with("#") || trimmed.is_empty() {
+            None
+        } else {
+            let mut iter = trimmed.split(":");
+            let key_str = iter.next().unwrap().trim();
+            let typed_key = Parameter::from_str(key_str)
```

Comment:
```suggestion
            let typed_key: Parameter = key_str.parse()
```

as a rule of thumb, never call from_str, always call parse